UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs

```diff
@@ -43,34 +45,54 @@ public class FulfilmentDataService : IFulfilmentDataService
             this.fulfilmentAncillaryFiles = fulfilmentAncillaryFiles;
             this.fulfilmentSalesCatalogueService = fulfilmentSalesCatalogueService;
             this.fulfilmentCallBackService = fulfilmentCallBackService;
+            this.monitorHelper = monitorHelper;
         }
 
         public async Task<string> CreateExchangeSet(SalesCatalogueServiceResponseQueueMessage message)
         {
+            DateTime createExchangeSetTaskStartAt = DateTime.UtcNow;
             string homeDirectoryPath = configuration["HOME"];
             var exchangeSetPath = Path.Combine(homeDirectoryPath, DateTime.UtcNow.ToString("ddMMMyyyy"), message.BatchId, fileShareServiceConfig.Value.ExchangeSetFileFolder);
             var exchangeSetRootPath = Path.Combine(exchangeSetPath, fileShareServiceConfig.Value.EncRoot);
             var exchangeSetPathForUploadZipFile = Path.Combine(homeDirectoryPath, DateTime.UtcNow.ToString("ddMMMyyyy"), message.BatchId);
             var listFulfilmentData = new List<FulfilmentDataResponse>();
-
+            DateTime downloadEncFilesFromFileShareServiceTaskStartAt = DateTime.UtcNow;
             var response = await DownloadSalesCatalogueResponse(message);
             if (response.Products != null && response.Products.Any())
             {
                 int parallelSearchTaskCount = fileShareServiceConfig.Value.ParallelSearchTaskCount;
                 int productGroupCount = response.Products.Count % parallelSearchTaskCount == 0 ? response.Products.Count / parallelSearchTaskCount : (response.Products.Count / parallelSearchTaskCount) + 1;
                 var productsList = CommonHelper.SplitList((response.Products), productGroupCount);
-
+                List<FulfilmentDataResponse> fulfilmentDataResponses = new List<FulfilmentDataResponse>();
+                object sync = new object();
+                int totalHitCountsToFileShareServiceForQuery = 0;                
                 var tasks = productsList.Select(async item =>
                 {
-                    listFulfilmentData.AddRange(await QueryAndDownloadFileShareServiceFiles(message, item, exchangeSetRootPath));
+                    fulfilmentDataResponses = await QueryAndDownloadFileShareServiceFiles(message, item, exchangeSetRootPath);
+                    int queryCount = fulfilmentDataResponses.Count > 0 ? fulfilmentDataResponses.FirstOrDefault().TotalHitCountForQueryFileShareService : 0;
+                    lock (sync)
+                    {
+                        totalHitCountsToFileShareServiceForQuery += queryCount;
+                    }
+                    listFulfilmentData.AddRange(fulfilmentDataResponses);
                 });
-                await Task.WhenAll(tasks);
+
+                await Task.WhenAll(tasks);                
+                DateTime downloadEncFilesFromFileShareServiceTaskCompletedAt = DateTime.UtcNow;
+                int folderCount = 0;
```

Comment:
Rename variable totalHitCountsToFileShareServiceForQuery at line 75 from to 
fileShareServiceSearchQueryCount

and variable folderCount at line 82 to downloadedENCFileCount

Also rename "Query and download file share service data" at line 87 to "Query and Download ENC Files Task"

Reply:
Done